File: app/src/main/java/com/botijasoftware/utils/interpolator/DecelerationInterpolator.java
package com.botijasoftware.utils.interpolator;

public class DecelerationInterpolator implements Interpolator {
	//public abstract Interpolator() {}
	public float interpolate(float v0, float v1, float t) {
		float a = 1-((1-t)*(1-t));
		return v0 + a*(v1 - v0);
	}
}

File: app/src/main/java/com/botijasoftware/utils/interpolator/DecelerationInterpolator.kt
package com.botijasoftware.utils.interpolator

class DecelerationInterpolator : Interpolator {
    //public abstract Interpolator() {}
    override fun interpolate(v0: Float, v1: Float, t: Float): Float {
        val a = 1 - (1 - t) * (1 - t)
        return v0 + a * (v1 - v0)
    }
}
